### DISCUSSION TITLE
Hotend max_temp???

### DISCUSSION THREAD

**Discussion by b-fitzpatrick** [xblax/flashforge_ad5m_klipper_mod#178] [General]
How is the extruder max_temp set? I see "max_temp: 350" in the [extruder] section of printer.base.cfg. I'm having to run much hotter temps with this printer to get similar results (270C for PETG, where I typically run 245 or 250C), so I suspect it may be reading higher than actual. Klipper shuts down somewhere around 290C. How does that work? I want to understand before I start messing with things.

**Comment by consp** [comment]
> Klipper shuts down somewhere around 290C

I think [this is preventing it](https://github.com/xblax/flashforge_ad5m_klipper_mod/blob/439fd0fa8ae83c4cfa5689ba0717bb24d4e14107/printer_configs/printer.base.cfg#L84C1-L88C17) but I'm not 100% sure.

Afaik the extruders are designed for max 280. I haven't had issues with it with the ABS I was using which requires 260c for good initial layer adhesion.

If it reports too high of a temp, it's probably a misconfigured temperature probe (which is a factory setting in [printer.base.cfg](https://github.com/xblax/flashforge_ad5m_klipper_mod/blob/439fd0fa8ae83c4cfa5689ba0717bb24d4e14107/printer_configs/printer.base.cfg#L65), e.g. wrong K value). That could skew it a bit at higher temperatures.

Could also be some added resistance somewhere, extruder connector maybe? Try to reseat the extruder and spray some deoxid on the contacts if possible. Don't know how the sensor is connected but might skew the results a bit at higher temperatures (Most likely lower but that depends on how it's configured).

If you have another probe (or another ntc) you could test it by attaching it to the extruder and measuring the resistance to verify it's not actually heating up as much as you expect. Don't know if the metal will allow IR thermometers to work well.

**Comment by xblax** [comment]
I don't observe temperature anomalies with the 5M. I can print PETG with 230° at 300m/s and 0.2 layer height. Temps are the same as with my previous printer.

You can try swapping the nozzle unit because the sensor should be in there. Maybe they send a replacement if you contact Flashforge support.

**Reply by KaruroChori** [reply]
Just a minor thing to consider, but I doubt it is at play here.
0.4  & 0.2mm nozzles are in copper/brass. 
0.6 and 0.8mm are hardened steel.
So, based on the target flow you might have to compensate the temperature a bit to make up for the limited thermal conductivity.

**Reply by b-fitzpatrick** [reply]
I'm still using the 0.4 mm nozzle, so material shouldn't be a factor. I have both a Pro and a non-Pro, so I swapped nozzles between the two to test.

I have a thermocouple I could user for verification, but I don't know how to take a good reading. With a standard hotend, I'd remove the nozzle and stick the thermocouple up inside the heater block.

**Reply by consp** [reply]
> but I don't know how to take a good reading

If you have a multi-meter you can measure the resistance and convert it to a temperature ([e.g. with this table](https://www.bapihvac.com/wp-content/uploads/2010/11/Thermistor_100K.pdf)). Assuming it's a 100k 3950 beta one.

**Reply by b-fitzpatrick** [reply]
I know how to take a measurement, just not a good one. I can hold it against the nozzle, but it reads low by 25-30C. I don't have confidence that is accurate, though. I would be more confident if I could get the thermocouple inside the heater block.

**Comment by Trogdor04** [comment]
Have we figured this out?  The ADC error calls out the "eboard".  I'm assuming that is the toolhead board.  Is there a way to access the cfg for the eboard?
